### DISCUSSION TITLE
Consider implementing stat functions

### DISCUSSION THREAD

**Discussion by springcomp** [jmespath-community/jmespath.spec#88] [Ideas]
In light of the proposal to include [arithmetic operations](https://github.com/jmespath-community/jmespath.spec/discussions/25), please consider introducing a range of useful statistical functions.

**Comment by innovate-invent** [comment]
What sort of functions?

Should we introduce functions that can be done with a minimal alternative?

avg: `([%].$+@)/length(@)` (already exists)
median: `[length(@)/2]` (assuming we allow dynamic subscription)
count: `[%{}].$+{@: ($[@] || 0) + 1}` (assuming we allow dynamic keys and subscription)

**Reply by springcomp** [reply]
That‘s the eternal debate between functions and syntax 🙂.
I actually did not had any functions in mind. I just initiated this idea at the same time as the math functions proposal.
